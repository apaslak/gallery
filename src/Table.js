import React from 'react'
import { Container, Row, Col } from 'react-grid-system';

const Table = (props) => {
  const _renderImage = (image) => {
    return (
      <Col key={image.id}>
        <img
          onClick={() => props.handleOpenModal(image)}
          src={image.url}
          className={image.className}
          alt={image.title}
        ></img>
      </Col>
    );
  }
  const _renderRow = (row, index) => {
    return (
      <Row key={index}>
        {row.map(image => _renderImage(image))}
      </Row>
    )
  }
  let data;

  if (props.rows) {
    data = props.rows.map((row, index) => _renderRow(row, index));
  } else {
    data = '';
  }

  return (
    <Container fluid>
      {data}
    </Container>
  )
};

export default Table;
