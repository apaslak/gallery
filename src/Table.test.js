import React from 'react';
import Table from './Table';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const component = renderer.create(
    <Table />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

it('renders data', () => {
  const rows = [
    [{ id: 1, url: 'https://abc.com', title: 'ABC' }],
    [{ id: 2, url: 'https://def.com', title: 'DEF' }]
  ];
  const component = renderer.create(
    <Table rows={rows}/>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

