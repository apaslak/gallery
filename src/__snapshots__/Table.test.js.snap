// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders data 1`] = `
<div
  style={
    Object {
      "boxSizing": "border-box",
      "marginLeft": "auto",
      "marginRight": "auto",
      "paddingLeft": "15px",
      "paddingRight": "15px",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "normal",
        "display": "flex",
        "flexGrow": 0,
        "flexShrink": 0,
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
        "marginLeft": "-15px",
        "marginRight": "-15px",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 0,
          "left": "auto",
          "marginLeft": "0%",
          "maxWidth": "100%",
          "minHeight": "1px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "position": "relative",
          "right": "auto",
          "width": "100%",
        }
      }
    >
      <img
        alt="ABC"
        onClick={[Function]}
        src="https://abc.com"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "alignItems": "normal",
        "display": "flex",
        "flexGrow": 0,
        "flexShrink": 0,
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
        "marginLeft": "-15px",
        "marginRight": "-15px",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 0,
          "left": "auto",
          "marginLeft": "0%",
          "maxWidth": "100%",
          "minHeight": "1px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "position": "relative",
          "right": "auto",
          "width": "100%",
        }
      }
    >
      <img
        alt="DEF"
        onClick={[Function]}
        src="https://def.com"
      />
    </div>
  </div>
  <span
    style={
      Object {
        "clear": "both",
        "display": "table",
      }
    }
  />
</div>
`;

exports[`renders without crashing 1`] = `
<div
  style={
    Object {
      "boxSizing": "border-box",
      "marginLeft": "auto",
      "marginRight": "auto",
      "paddingLeft": "15px",
      "paddingRight": "15px",
      "position": "relative",
    }
  }
>
  
  <span
    style={
      Object {
        "clear": "both",
        "display": "table",
      }
    }
  />
</div>
`;
